nym-connect/desktop/src/context/main.tsx

```diff
@@ -162,39 +167,60 @@ export const ClientContextProvider: FCWithChildren = ({ children }) => {
   const shouldUseUserGateway = !!userDefinedGateway.gateway && userDefinedGateway.isActive;
   const shouldUseUserSP = !!userDefinedSPAddress.address && userDefinedSPAddress.isActive;
 
-  const setServiceProvider = async (newServiceProvider: ServiceProvider) => {
-    await invoke('set_gateway', {
-      gateway: shouldUseUserGateway ? userDefinedGateway.gateway : newServiceProvider.gateway,
-    });
+  const applyServiceProvider = async (newServiceProvider: ServiceProvider) => {
     await invoke('set_service_provider', {
       serviceProvider: shouldUseUserSP ? userDefinedSPAddress.address : newServiceProvider.address,
     });
   };
 
+  const applyGateway = async (newGateway: Gateway) => {
+    await invoke('set_gateway', {
+      gateway: shouldUseUserGateway ? userDefinedGateway.gateway : newGateway.identity,
+    });
+  };
+
   const getRandomSPFromList = (services: ServiceProvider[]) => {
     const randomSelection = services[Math.floor(Math.random() * services.length)];
     return randomSelection;
   };
 
+  const getRandomGatewayFromList = (theGateways: Gateway[]) => {
+    const randomSelection = theGateways[Math.floor(Math.random() * theGateways.length)];
+    return randomSelection;
+  };
```

Comment:
Function declaration for one code line which is called only once is redundant

Extra function declaration could be avoided and `theGateways[Math.floor(Math.random() * theGateways.length)]` could be moved back in the calling code

Same for `services[Math.floor(Math.random() * services.length)]`